lib/Promise.php

```diff
@@ -188,16 +188,21 @@ public function wait()
             return $this->value;
         } else {
             // If we got here, it means that the asynchronous operation
-            // errored. Therefore we need to throw an exception.
-            throw $this->value;
+            // errored. Therefore, we need to throw an exception.
+            if ($this->value instanceof \Throwable) {
+                throw $this->value;
+            }
+            // The state should have been REJECTED, with "value" a Throwable
+            // But "value" was not a Throwable. So throw a more general exception.
+            throw new \LogicException('The Promise was not fulfilled but no exception was specified');
```

Comment:
phpstan is confident that "value" will be a Throwable at this point. So I added an extra check, and throw a LogicException if value was not a Throwable. LogicException is already used elsewhere in this code, so I chose to throw that.